Core/Rule.cs

```diff
@@ -1,15 +1,18 @@
 ﻿namespace Core
 {
-    public class Rule
+    public readonly struct Rule
```

Comment:
Codacy found an issue: [Implement 'IEquatable<T>' in value type 'Rule'.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=5924375)